tools/publish_aws.py

```diff
@@ -38,81 +38,56 @@ def __init__(
         self._pkg_version = package_version
         self._input_dir_path = input_dir_path
 
-        self._aws_region = os.environ.get('AWS_UPLOAD_REGION', '')
         self._universe_url_prefix = os.environ.get(
             'UNIVERSE_URL_PREFIX',
             'https://universe-converter.mesosphere.com/transform?url=')
+
+        if not os.path.isdir(input_dir_path):
+            raise Exception('Provided package path is not a directory: {}'.format(input_dir_path))
+
+        self._artifact_paths = []
+        for artifact_path in artifact_paths:
+            if not os.path.isfile(artifact_path):
+                err = 'Provided package path is not a file: {} (full list: {})'.format(artifact_path, artifact_paths)
+                raise Exception(err)
+            self._artifact_paths.append(artifact_path)
+
         s3_bucket = os.environ.get('S3_BUCKET')
         if not s3_bucket:
             s3_bucket = 'infinity-artifacts'
         logger.info('Using artifact bucket: {}'.format(s3_bucket))
+
         s3_dir_path = os.environ.get('S3_DIR_PATH', 'autodelete7d')
         dir_name = '{}-{}'.format(
             time.strftime("%Y%m%d-%H%M%S"),
             ''.join([random.SystemRandom().choice(string.ascii_letters + string.digits) for i in range(16)]))
 
         # sample s3_directory: 'infinity-artifacts/autodelete7d/kafka/20160815-134747-S6vxd0gRQBw43NNy'
-        self._s3_directory = os.environ.get(
+        s3_directory_url = os.environ.get(
             'S3_URL',
             's3://{}/{}/{}/{}'.format(
                 s3_bucket,
                 s3_dir_path,
-                self._pkg_name,
+                package_name,
                 dir_name))
+        self._uploader = universe.S3Uploader(self._pkg_name, s3_directory_url, self._dry_run)
```

Comment:
A bit strange that we use `package_name` above and below and `self._pkg_name` here.